### PR TITLE
Merging 'feature/models-relationship' to 'develop'

### PR DESCRIPTION
**Pull request by andela-nmartin**
Hi guys, these are the database, models declaration and relationship breaking changes made:

1. server/config/db-connect.js
    - added DB authentication to report on a successful connection  
    - removed the err argument passed on the success promise callback

2. Models are now use sequelize importation instead of being required as individuals modules which could create individual connections to the DB for everywhere they are required.

3. Models are now instantiated once in the `root index.js` file and added to the express application instance so as to be singleton-esque. A single connection is created and the same models then persisted across the applications

4. Models relationships are defined in their own file: `relationships.js` in the models folder. Also instantiated in the models `index.js`

5. In controllers or anywhere required, you can now access the models via `req.app.get(‘models’).ModelName` (1st letter is uppercase)

6. Added database port declaration to the application config, referenced from the env file

### CODE REVIEW

File: server/controllers/users.js [hunk 1]
```diff
@@ -75,6 +75,7 @@
 
     // Middleware to get all users
     all: function(req, res) {
+      var User = req.app.get('models').Users;
```
Comment (andela-gukpere) [inline server/controllers/users.js]: **Code Quality - Nitpick**

Other model imports are plural, this should follow the convention as well.
Is there a reason its `User` not `Users`?

File: server/controllers/users.js [hunk 2]
```diff
@@ -47,8 +47,8 @@
     authenticate: function(req, res, next) {
```
Comment (andela-gukpere) [inline server/controllers/users.js]: Refer to https://github.com/andela/vvida/commit/a5169f1356bd0083bb34cabf99a4ad12fdd30974#commitcomment-14071696

File: server/models/events.js [hunk 3]
```diff
@@ -1,52 +1,49 @@
...
+      //   allowNull: false,
+      //   references: {
+      //     model: users,
+      //     key: 'id'
+      //   }
+      // },
 
       // event name
       // hold the name of the events
       ev_name: {
```
Comment (andela-gukpere) [inline server/models/events.js]: **Code Quality - Critical**

Why do we have `ev_name`?
When you have a name field to hold text, the name in an `Events` table, the field should be called `name`, the `ev_` prefix serves no purpose.

@andela-nmartin @Andela-thomas @kn9ts

File: server/models/items.js [hunk 4]
```diff
@@ -1,42 +1,38 @@
...
       // references the user id in users table
-      user_id: {
-        type: Seq.INTEGER,
-        allowNull: false,
-        references: {
-          model: users,
-          key: 'id'
-        }
-      },
+      // user_id: {
```
Comment (andela-gukpere) [inline server/models/items.js]: **Code Quality - Critical**

We should remove all commented code blocks that are no longer in use.

File: server/models/promotions.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+module.exports = function(sequelize, DataType) {
+  return sequelize.define('ItemPromotions', {
+      // FOREIGN KEY
```
Comment (andela-gukpere) [inline server/models/promotions.js]: Refer to https://github.com/andela/vvida/pull/89/files#r43408247

File: server/services/auth/google.auth.js [hunk 6]
```diff
@@ -1,10 +1,9 @@
-var User = require('../../models/users');
 // Use the GoogleStrategy within Passport.
 // Strategies in Passport require a `verify` function, which accept
 // credentials (in this case, an accessToken, refreshToken, and Google
 // profile), and invoke a callback with a user object.
-module.exports = function(passport, GoogleStrategy, config) {
-
+module.exports = function(app, passport, GoogleStrategy, config) {
+  var User = app.get('models').Users;
```
Comment (andela-gukpere) [inline server/services/auth/google.auth.js]: Refer to https://github.com/andela/vvida/pull/89/files#r43381894

**Comment by andela-gukpere** [comment]
We should make these adjustments in the scripts they appear in.
There's a lot, I can't see it all.

File: server/models/items.js [hunk 7]
```diff
@@ -1,42 +1,16 @@
...
+      name: {
+        type: DataType.STRING,
         allowNull: false,
       },
       // item description
       // gives a detailed description of what the item is all about
 
-      item_desc: {
-        type: Seq.TEXT,
+      desc: {
```
Comment (andela-gukpere) [inline server/models/items.js]: Can we have this as `description`.
There's no need to abreviate.

File: server/controllers/users.js [hunk 8]
```diff
@@ -101,8 +102,10 @@
 
     // Middleware to get users by ID
     find: function(req, res) {
-      var userId = req.params.id;
-      User.findOne({
+      var Users = req.app.get('models').Users,
```
Comment (andela-gukpere) [inline server/controllers/users.js]: Apply this fix to https://github.com/andela/vvida/pull/89/files#r43381894 as well.

File: server/models/images.js [hunk 9]
```diff
@@ -1,35 +1,9 @@
-//var require the seqalize module
-var Seq = require('sequelize'),
-  db = require('../config/db-connect'),
-  users = require('./users'),
-  items = require('./items'),
-  images = db.define('images', {
-
-      // FOREIGN KEY
-      // references the item id in the items table
-      item_id: {
```
Comment (andela-gukpere) [inline server/models/images.js]: The Item ID is used to refer to the item that the image belongs to.
Was the relationship referenced somewhere else?

File: server/models/relationships.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  m.Items.belongsTo(m.Categories);
+  m.Items.belongsTo(m.Users);
+
+  // Item promotions: recommedations, shares and sponsors
+  m.Promotions.belongsTo(m.Users);
+
+  // Messaging
+  m.Messages.belongsTo(m.Users);
+  m.Messages.belongsTo(m.Users, {
+    foreignKey: 'receiver_id'
```
Comment (andela-gukpere) [inline server/models/relationships.js]: Does this replace the original reference in the `messages` model?

File: server/services/auth/google.auth.js [hunk 11]
```diff
@@ -1,20 +1,19 @@
...
+module.exports = function(app, passport, GoogleStrategy, config) {
+  var Users = app.get('models').Users;
   passport.use(new GoogleStrategy(config.auth.GOOGLE,
     function(accessToken, refreshToken, profile, done) {
       console.log(profile);
 
       // make the code asynchronous
-      // User.findOne won't fire until we have all our data back from Google
+      // Users.findOne won't fire until we have all our data back from Google
       process.nextTick(function() {
```
Comment (andela-gukpere) [inline server/services/auth/google.auth.js]: Why do we need `process.nextTick`.

Comment (kn9ts) [inline server/services/auth/google.auth.js]: Keep the callback truelly asynchronous. That's all I remember. :)

Comment (andela-gukpere) [inline server/services/auth/google.auth.js]: :laughing: truly?
Async++?

Comment (kn9ts) [inline server/services/auth/google.auth.js]: Yes! I do probably need it. If I fall into errors then I will know I need it.

Comment (andela-gukpere) [inline server/services/auth/google.auth.js]: Any progress yet?
_Not happy I have to ask this question_
**#Engagement**

File: server/models/items.js (hunk 7 above)
Comment (kn9ts) [inline server/models/items.js]: Yes. True.

File: server/models/relationships.js (hunk 10 above)
Comment (kn9ts) [inline server/models/relationships.js]: Yes it does. Along with all the relationships of all the models associated with each other.

File: server/services/auth/google.auth.js (hunk 11 above)
Comment (kn9ts) [inline server/services/auth/google.auth.js]: My plan is to handle the changes in a `refactor` branch. Could have done these changes yesterday but I was not available for the rest of the day.

File: server/models/images.js (hunk 9 above)
Comment (kn9ts) [inline server/models/images.js]: Yes it is. In the `relationships.js` relationships module file

**Pull request merged**